skbio/alignment/_tabular_msa.py

```diff
@@ -1095,6 +1095,252 @@ def _assert_valid_sequences(self, sequences):
+        """Join this MSA with another by sequence (horizontally).
+
+        Sequences will be joined by index labels. MSA ``positional_metadata``
+        will be joined by columns. Use `how` to control join behavior.
+
+        Alignment is **not** recomputed during join operation (see *Notes*
+        section for details).
+
+        Parameters
+        ----------
+        other : TabularMSA
+            MSA to join with. Must have same ``dtype`` as this MSA.
+        how : {'strict', 'inner', 'outer', 'left', 'right'}, optional
+            How to join the sequences and MSA `positional_metadata`:
+
+            * ``'strict'``: MSA indexes and `positional_metadata` columns must
+              match
+
+            * ``'inner'``: an inner-join of the MSA indexes and
+              ``positional_metadata`` columns (only the shared set of index
+              labels and columns are used)
+
+            * ``'outer'``: an outer-join of the MSA indexes and
+              ``positional_metadata`` columns (all index labels and columns are
+              used). Unshared sequences will be padded with the MSA's default
+              gap character (``TabularMSA.dtype.default_gap_char``). Unshared
+              columns will be padded with NaN.
+
+            * ``'left'``: a left-outer-join of the MSA indexes and
+              ``positional_metadata`` columns (this MSA's index labels and
+              columns are used). Padding of unshared data is handled the same
+              as ``'outer'``.
+
+            * ``'right'``: a right-outer-join of the MSA indexes and
+              ``positional_metadata`` columns (`other` index labels and columns
+              are used). Padding of unshared data is handled the same as
+              ``'outer'``.
+
+        Returns
+        -------
+        TabularMSA
+            Joined MSA. There is no guaranteed ordering to its index (call
+            ``sort`` to define one).
+
+        Raises
+        ------
+        ValueError
+            If `how` is invalid.
+        ValueError
+            If either the index of this MSA or the index of `other` contains
+            duplicates.
+        ValueError
+            If ``how='strict'`` and this MSA's index doesn't match with
+            `other`.
+        ValueError
+            If ``how='strict'`` and this MSA's ``positional_metadata`` columns
+            don't match with `other`.
+        TypeError
+            If `other` is not a subclass of ``TabularMSA``.
+        TypeError
+            If the ``dtype`` of `other` does not match this MSA's ``dtype``.
+
+        See Also
+        --------
+        extend
+        sort
+        skbio.sequence.Sequence.concat
+
+        Notes
+        -----
+        The join operation does not automatically perform re-alignment;
+        sequences are simply joined together. Therefore, this operation is not
+        necessarily meaningful on its own.
+
+        The index of this MSA and the index of `other` must each be unique;
```

Comment:
Add "respectively"? The wording as is could be construed as no matching labels between either index